test/kubernetes_test.go

```diff
@@ -12,63 +11,158 @@ import (
-	{"mynginx.any.svc.coredns.local.", 1, 1},                     // One A record, via wildcard namespace
-	{"someservicethatdoesnotexist.*.svc.coredns.local.", 0, 0},   // Record does not exist with wildcard for namespace
-	{"someservicethatdoesnotexist.any.svc.coredns.local.", 0, 0}, // Record does not exist with wildcard for namespace
-	{"*.demo.svc.coredns.local.", 2, 2},                          // Two A records, via wildcard
-	{"any.demo.svc.coredns.local.", 2, 2},                        // Two A records, via wildcard
-	{"*.test.svc.coredns.local.", 0, 0},                          // Two A record, via wildcard that is not exposed
-	{"any.test.svc.coredns.local.", 0, 0},                        // Two A record, via wildcard that is not exposed
-	{"*.*.svc.coredns.local.", 2, 2},                             // Two A records, via namespace and service wildcard
+func makeRR(s string) dns.RR {
+	rr, err := dns.NewRR(s)
+	if err != nil {
+		log.Printf("Bad test RR (%s): %s", s, err)
+	}
+	return rr
 }
 
-// Test data for SRV records
-var testdataLookupSRV = []struct {
-	Query            string
-	TotalAnswerCount int
-	//	ARecordCount     int
-	SRVRecordCount int
-}{
-	// Matching queries
-	{"mynginx.demo.svc.coredns.local.", 1, 1}, // One SRV record, should exist
-
-	// Failure queries
-	{"mynginx.test.svc.coredns.local.", 0, 0},                     // One SRV record, is not exposed
-	{"someservicethatdoesnotexist.demo.svc.coredns.local.", 0, 0}, // Record does not exist
-
-	// Namespace wildcards
-	{"mynginx.*.svc.coredns.local.", 1, 1},                       // One SRV record, via wildcard namespace
-	{"mynginx.any.svc.coredns.local.", 1, 1},                     // One SRV record, via wildcard namespace
-	{"someservicethatdoesnotexist.*.svc.coredns.local.", 0, 0},   // Record does not exist with wildcard for namespace
-	{"someservicethatdoesnotexist.any.svc.coredns.local.", 0, 0}, // Record does not exist with wildcard for namespace
-	{"*.demo.svc.coredns.local.", 2, 2},                          // Two (mynginx, webserver) SRV record, via wildcard
-	{"any.demo.svc.coredns.local.", 2, 2},                        // Two (mynginx, webserver) SRV record, via wildcard
-	{"*.test.svc.coredns.local.", 0, 0},                          // One SRV record, via wildcard that is not exposed
-	{"any.test.svc.coredns.local.", 0, 0},                        // One SRV record, via wildcard that is not exposed
-	{"*.*.svc.coredns.local.", 2, 2},                             // Two SRV record, via namespace and service wildcard
+type testquery struct {
+	fqdn  string
+	qtype uint16
 }
 
-func TestKubernetesIntegration(t *testing.T) {
+type testanswer struct {
+	rcode int
+	rrs   []dns.RR
+}
+
+type testdefn struct {
+	query  testquery
+	answer testanswer
+}
 
-	// t.Skip("Skip Kubernetes Integration tests")
-	// subtests here (Go 1.7 feature).
-	testLookupA(t)
-	testLookupSRV(t)
+// Test data
+// TODO: Fix the actual RR values
+var testdata = []testdefn{
```

Comment:
There is also infra for this - look at the other tests (can remember from the top of my head which one)